### DISCUSSION TITLE
Amt not configuring

### DISCUSSION THREAD

**Discussion by laramy2020** [Ylianst/MeshCentral#6343] [Q&A]
I have a dell Latitude 5320, it only has wifi, but it's bios claims network passthrough to thunderbolt which i have enabled and plugged in.

I have tried both the agent and the usb configuration and amt will not configure

./meshcmd amtinfo
```
Intel AMT v15.0.45, pre-provisioning state.
Wireless Disabled, DHCP, 00:00:00:00:00:00
Connection Status: Unknown, CIRA: Disconnected.
```

Os is a fully up-to-date arch system

/dev/mei0 exists

**Comment by laramy2020** [comment]
oh here is the meshcentral config
![image](https://github.com/user-attachments/assets/81e3575b-ba90-4f11-a128-75f77cc63c7b)

**Reply by si458** [reply]
If you have the agent installed on the device, boot into windows, then in the console tab of the device, type `amtconfig` and watch its output for some debug info

It might explain why it won't activate or activated but wifi only

**Reply by laramy2020** [reply]
any way to run this in linux? only have one drive in it, dont feel like backing up or opening up disks and swaping out if i dont have to. i have meshcmd there but amtconfig is asking for a url and an id i tried the desktop

```bash
[root@box Downloads]# ./meshcmd amtconfig
No MeshCentral server URL specified, use --url [url].
[root@box Downloads]# ./meshcmd amtconfig --url mc.website.com
No device group identifier specified, use --id [identifier].
[root@box Downloads]# ./meshcmd amtconfig --url mc.website.com --id Desktops
Setting up MEI...
Starting Intel AMT configuration...
Started APF tunnel...
```

It just hangs here, i can ctrl+c it

**Reply by laramy2020** [reply]
and 4433 is enabled and visible, and it is using the self signed cert that the video guide says it uses.

**Reply by si458** [reply]
No no sorry,
1. Install the meshagent on the remote computer
2. Go to the device in meshcentral web ui
3. Click the console tab of the device
4. Type in `amtconfig` in there and watch it's output

U only need the meshcmd if u want to provision on the device without meshcentral for example

**Reply by laramy2020** [reply]
ok that is weird, it says this

`Intel AMT not detected`

**Reply by si458** [reply]
Right, this is possibly because it's linux,
So I think you need to check ur kernel has the amt drivers installed and activate

**Reply by laramy2020** [reply]
it looks to be active, otherwise /dev/mei0 should not show nor would this show an output 

https://github.com/mjg59/mei-amt-check

```
sudo ./mei-amt-check
AMT present: true
AMT provisioning state: unprovisioned
Flash:  15.0.45
Netstack:       15.0.45
AMTApps:        15.0.45
AMT:    15.0.45
Sku:    8
VendorID:       8086
Build Number:   2411
Recovery Version:       15.0.45
Recovery Build Num:     2411
Legacy Mode:    False
```

**Reply by laramy2020** [reply]
oh it might be because wsman and wsmancli needs to be installed and it's source is not building from the aur in arch. looking into it

**Reply by laramy2020** [reply]
ok i got wsman installed and running, but it seems to have done nothing. i am at a loss of where else i should look

**Reply by laramy2020** [reply]
```
00:16.0 Communication controller [0780]: Intel Corporation Tiger Lake-LP Management Engine Interface [8086:a0e0] (rev 20)
Subsystem: Dell Device [1028:0a1f]
Kernel driver in use: mei_me
Kernel modules: mei_me
```
it looks like the kernel module is loaded

**Reply by si458** [reply]
And what happens now if u restart the Linux machine then run the amtconfig from the console tab of the device? 
Is it still the same error saying no amt?

**Reply by laramy2020** [reply]
still says no amt

**Reply by laramy2020** [reply]
you know i did search the issues before, i guess my search terms where bad #3846

**Reply by laramy2020** [reply]
![image](https://github.com/user-attachments/assets/098f466a-4995-40b3-9ab6-5fd9e5c6bff9)
![image](https://github.com/user-attachments/assets/23daa862-633a-4b03-85a1-1f6067bd2dd2)

i created an amt only area, it shows up but does not allow configure if i do it via the webui

If i try with meshcmd commands it then asks for credentials, and i give it the credentials i made and it just gets stuck on trying credentals

![image](https://github.com/user-attachments/assets/fb0c0de0-0492-4fb3-8abf-9148fb02aa53)
![image](https://github.com/user-attachments/assets/1c37187d-e2a0-42bb-971d-b8d97c45785c)


i do have an ! in my password, could that also cause issues?

**Reply by si458** [reply]
@laramy2020 Sadly, im out of ideas, also as the other issue explaines, all AMT issues are on hold because we have no AMT development/developer

Without access to hardware, we can't fully look into issues :(
 
I only have 2 devices myself, an AMT V7, and AMT v12 on 2 different dell machines (6th processors), both have Windows installed so can't just uninstall and reinstall windows because the amt provisioning is separate!

If anyone is willing to donate money or hardware, I do have a donation page setup https://www.si458.co.uk/2024/01/05/donation/
